compliance/sparql/src/test/java/org/eclipse/rdf4j/query/parser/sparql/function/CustomAggregateFunctionEvaluationTest.java

```diff
@@ -159,13 +157,102 @@ public void testCustomFunction_MultipleSumWithDistinct() {
 			try (TupleQueryResult result = conn.prepareTupleQuery(query).evaluate()) {
 				BindingSet bs = result.next();
 				assertThat(bs.getValue("m").stringValue()).isEqualTo("62.390000200001");
-				assertThat(bs.getValue("sa").stringValue()).isEqualTo("96.690000200001");
+				assertThat(bs.getValue("sa").stringValue()).isEqualTo("125.933564200001");
 				assertThat(bs.getValue("m").stringValue()).isNotEqualTo(bs.getValue("sa").stringValue());
 				assertThat(result.hasNext()).isFalse();
 			}
 		}
 	}
 
+	@Test
+	public void testCustomFunction_MultipleSumWithDistinctGroupBy() {
+		String query = "select ?s (<" + functionFactory.getIri() + ">(distinct ?o) as ?m) where { \n"
+
+				+ "\t?s <urn:n> ?o . filter(?o > 0) } group by ?s order by ?s";
+		try (RepositoryConnection conn = rep.getConnection()) {
+			try (TupleQueryResult result = conn.prepareTupleQuery(query).evaluate()) {
+				assertThat(result.next().getValue("m").stringValue()).isEqualTo("12.5");
+				assertThat(result.next().getValue("m").stringValue()).isEqualTo("6");
+				assertThat(result.next().getValue("m").stringValue()).isEqualTo("12.5");
+				assertThat(result.next().getValue("m").stringValue()).isEqualTo("31.3");
+				assertThat(result.next().getValue("m").stringValue()).isEqualTo("0.090000200001");
+				assertThat(result.next().getValue("m").stringValue()).isEqualTo("60.543564");
+				assertThat(result.next().getValue("m").stringValue()).isEqualTo("3");
+				assertThat(result.hasNext()).isFalse();
+			}
+		}
+	}
+
+	@Test
+	public void testCustomFunction_MultipleSumWithHaving() {
+		String query = "select ?s (<" + functionFactory.getIri() + ">(?o) as ?m) where { \n"
+				+ "\t?s <urn:n> ?o . }\n" +
+				"GROUP BY ?s \n" +
+				"HAVING((<" + functionFactory.getIri() + ">( ?o)) > 60)";
+		try (RepositoryConnection conn = rep.getConnection()) {
+			TupleQuery tupleQuery = conn.prepareTupleQuery(query);
+			try (TupleQueryResult result = tupleQuery.evaluate()) {
+				assertThat(result.next().getValue("s").stringValue()).isEqualTo("http://example/book7");
+				assertThat(result.hasNext()).isFalse();
+			}
+		}
+	}
```

Comment:
When comparing the tupleQuery from this test with the one using the built in SUM function (the test below) there seems to be some differences.